src/plugins/elasticsearch/index.js

```diff
@@ -24,7 +24,7 @@ module.exports = function (kibana) {
           key: Joi.string()
         }).default(),
         apiVersion: Joi.string().default('2.0'),
```

Comment:
So the question for anyone looking at this is: do we couple upgrading our elasticsearch js module with this minimum version change or treat them as two separate things?  Personally, I think we should treat them separately.